external-crates/move/crates/move-compiler/tests/move_check/parser/vector_literal_unclosed_type_args.exp

```diff
@@ -1,9 +1,34 @@
 error[E01002]: unexpected token
-  ┌─ tests/move_check/parser/vector_literal_unclosed_type_args.move:4:27
+  ┌─ tests/move_check/parser/vector_literal_unclosed_type_args.move:4:28
   │
-4 │         let v = vector<u64[0, 1];
-  │                       -   ^ Expected '>'
-  │                       │    
-  │                       To match this '<'
-  │                       Perhaps you need a blank space before this '<' operator?
+4 │         let _v = vector<u64[0, 1];
+  │                            ^
+  │                            │
+  │                            Unexpected '['
+  │                            Expected ','
+
+error[E01002]: unexpected token
+  ┌─ tests/move_check/parser/vector_literal_unclosed_type_args.move:4:33
+  │
+4 │         let _v = vector<u64[0, 1];
+  │                                -^ Unexpected ']'
+  │                                │ 
+  │                                Expected '::' after an address in a module access chain
+
+error[E01002]: unexpected token
+  ┌─ tests/move_check/parser/vector_literal_unclosed_type_args.move:4:34
+  │
+4 │         let _v = vector<u64[0, 1];
+  │                        -         ^ Expected '>'
+  │                        │          
+  │                        To match this '<'
+
+error[E01002]: unexpected token
+  ┌─ tests/move_check/parser/vector_literal_unclosed_type_args.move:4:34
+  │
+4 │         let _v = vector<u64[0, 1];
+  │                                  ^
+  │                                  │
+  │                                  Unexpected ';'
+  │                                  Expected '['
```

Comment:
I'm confused about this set. If you put `[` in the follow set when parsing the type, does the error get better?

Reply:
Good eye! Adding `[` helps, but at the same time adding just `[` seems a bit ad-hoc. I experimented a bit on what could/should be a stop set for types (and type params) and first I went all they way in and added to it all tokens that should not be part of a type. This appeared too much, though, causing additional (unnecessary) diags being generated for the following tests:
- [move_check/parser/less_than_space.move](https://github.com/MystenLabs/sui/blob/main/external-crates/move/crates/move-compiler/tests/move_check/parser/less_than_space.move) where `return` being in the stop set causes extra diag as we bail out on it and start parsing again too early
- [move_2024/parser/positional_struct_fields_non_type_field.move](https://github.com/MystenLabs/sui/blob/main/external-crates/move/crates/move-compiler/tests/move_2024/parser/positional_struct_fields_non_type_field.exp)  where `type`being in the stop set causes extra diags for the same reason as described above

Ultimately then, I settled on the "type stop set" to include various punctuation (and similar) characters.